injection/sharedmain/main.go

```diff
@@ -165,6 +165,8 @@ func toControllerConstructors(namedCtors []injection.NamedControllerConstructor)
 // MainWithContext runs the generic main flow for controllers and
 // webhooks. Use MainWithContext if you do not need to serve webhooks.
 func MainWithContext(ctx context.Context, component string, ctors ...injection.ControllerConstructor) {
+	// Allow configuration of threads per controller
+	flag.IntVar(&controller.DefaultThreadsPerController, "threads-per-controller", controller.DefaultThreadsPerController, "Threads (goroutines) to create per controller")
```

Comment:
Not a huge fan of flags, there is a config map that the control can pull config out of. Maybe that is a better place?

Reply:
Hi @n3wscott, thank you for the feedback. I can see how the leader election configuration is read from a ConfigMap here, https://github.com/knative/pkg/blob/main/injection/sharedmain/main.go#L88-L104. But looking through the names of the existing ConfigMaps, I do not see a good candidate. Are you suggesting to introduce an additional ConfigmMap ? And if yes, how should we name it ? Would it only contain the threads-per-controller, or do you envision that related things that are today configurable using flags, like qps and burst, would be in there as well ? So maybe a `config-tuning` ?

Reply:
Ok good point. Perhaps then we could introduce an ENV VAR instead of a flag? something like `K_THREADS_PER_CONTROLLER` or something similar?

Reply:
Okay, as you wish. I put it at the same location. Let me know if you want it elsewhere.